Updates to internal GitLab Omnibus processes

jasonblais opened the pull request:
Not urgent, did some cleanup, fine to review next week :)

Changes:
1 - Updated testing section with correct steps
2 - Moved useful commands to their own section in our support handbook

Note: Not sure if the documented upgrade process is still accurate, Harrison?

justinegeffen commented on source/process/gitlab-process.rst:
```diff
@@ -29,10 +29,15 @@ If the release contains a security update, PM owner emails @marin and @briann in
-1. Each Mattermost version is tested on a GitLab Omnibus build at `http://ci-linux-gitlab-omnibus.mattermost.com/ <http://ci-linux-gitlab-omnibus.mattermost.com/>`__. Testing covers all core Mattermost features, including notifications and GitLab SSO.
-2. Before each merge request to GitLab Omnibus, upgrade is tested `following the steps below <https://docs.mattermost.com/process/gitlab-process.html#testing-upgrade-process>`__, using the `nightly Omnibus packages <https://packages.gitlab.com/gitlab/nightly-builds>`__ to validate the integration. This is so the packaging code and OAuth setup can be tested as well, which has historically been the ...
+1. Pull the latest master build of GitLab Omnibus, using the `nightly Omnibus packages <https://packages.gitlab.com/gitlab/nightly-builds>`_.
+2. Update the Mattermost version in `config/softwater/mattermost.rb`.
```
Where is the Mattermost version found?

jasonblais replied:
@hmhealey Can you help with this one?

hmhealey replied:
Do you mean where is it found in the file? I don't know how to style this properly in RST, but at the top of the file, you'll find a section that looks like
```
        name 'mattermost'
        default_version '5.19.0-rc2'

        source url: "https://releases.mattermost.com/#{version}/mattermost-team-#{version}-linux-amd64.tar.gz",
               md5: 'fe49d764324334480b296fc6c500b706'
```
You'll want to update the `default_version` and `md5` variables to the new versions.

justinegeffen commented on source/process/gitlab-process.rst:
```diff
@@ -29,10 +29,15 @@ If the release contains a security update, PM owner emails @marin and @briann in
-2. Before each merge request to GitLab Omnibus, upgrade is tested `following the steps below <https://docs.mattermost.com/process/gitlab-process.html#testing-upgrade-process>`__, using the `nightly Omnibus packages <https://packages.gitlab.com/gitlab/nightly-builds>`__ to validate the integration. This is so the packaging code and OAuth setup can be tested as well, which has historically been the ...
+1. Pull the latest master build of GitLab Omnibus, using the `nightly Omnibus packages <https://packages.gitlab.com/gitlab/nightly-builds>`_.
+2. Update the Mattermost version in `config/softwater/mattermost.rb`.
+3. Update the version in the GitLab documentation and add a changelog entry.
```
Version of the documentation itself, or the version found in the documentation - and is it the GitLab version that's being updated?

justinegeffen commented on source/process/gitlab-process.rst:
```diff
@@ -29,10 +29,15 @@ If the release contains a security update, PM owner emails @marin and @briann in
+2. Update the Mattermost version in `config/softwater/mattermost.rb`.
+3. Update the version in the GitLab documentation and add a changelog entry.
+4. Build GitLab Omnibus.
+5. Install the generated deb file on a local test server.
```
```suggestion
5. Install the generated `.deb` file on a local test server.
```

justinegeffen commented on source/process/gitlab-process.rst:
```diff
@@ -29,10 +29,15 @@ If the release contains a security update, PM owner emails @marin and @briann in
+4. Build GitLab Omnibus.
+5. Install the generated deb file on a local test server.
+
+The following enables the Mattermost team to test the upgrade process and validate the packaging code and OAuth setup which have historically been the main source of issues. Other test areas include:
```
```suggestion
The Mattermost team then test the upgrade process, and validate the packaging code and OAuth setup which have historically been the main source of issues. Other test areas include:
```

justinegeffen commented on source/process/gitlab-process.rst:
```diff
@@ -29,10 +29,15 @@ If the release contains a security update, PM owner emails @marin and @briann in
+
+The following enables the Mattermost team to test the upgrade process and validate the packaging code and OAuth setup which have historically been the main source of issues. Other test areas include:
 
  - Pre-provisioning OAuth configuration automatically on the Omnibus package
```
```suggestion
 - Pre-provisioning OAuth configuration automatically on the Omnibus package.
```

justinegeffen commented on source/process/gitlab-process.rst:
```diff
@@ -29,10 +29,15 @@ If the release contains a security update, PM owner emails @marin and @briann in
+The following enables the Mattermost team to test the upgrade process and validate the packaging code and OAuth setup which have historically been the main source of issues. Other test areas include:
 
  - Pre-provisioning OAuth configuration automatically on the Omnibus package
  - Mattermost ChatOps slash command integration
```
```suggestion
 - Mattermost ChatOps slash command integration.
```

justinegeffen commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
 
 #### Other debugging information
```
```suggestion
#### Other Debugging Information
```

justinegeffen commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
-- `gitlab-ctl <stop/start/restart/status/tail> <nginx/postgresql/redis/etc>` - Control, view status, or view logs of a different service.
-- `sudo gitlab-psql -d mattermost_production` - Access the embedded Mattermost database
+Useful commands:
+- `sudo /opt/gitlab/bin/gitlab-ctl reconfigure` - Update config.json and other configuration files, then restart all services (including Mattermost).
```
```suggestion
- `sudo /opt/gitlab/bin/gitlab-ctl reconfigure` - Update `config.json` and other configuration files, then restart all services (including Mattermost).
```

justinegeffen commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
+Useful commands:
+- `sudo /opt/gitlab/bin/gitlab-ctl reconfigure` - Update config.json and other configuration files, then restart all services (including Mattermost).
+- `sudo /opt/gitlab/bin/gitlab-ctl <stop/start/restart> mattermost` - Stop/start/restart Mattermost.
+- `sudo /opt/gitlab/bin/gitlab-ctl status mattermost` - Check if Mattermost is running. The message printed will start with `run` if it's running or `down` if it's not.
```
```suggestion
- `sudo /opt/gitlab/bin/gitlab-ctl status mattermost` - Check if Mattermost is running. The printed output will start with `run` if it's running or `down` if it's not.
```

justinegeffen commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
+- `sudo /opt/gitlab/bin/gitlab-ctl reconfigure` - Update config.json and other configuration files, then restart all services (including Mattermost).
+- `sudo /opt/gitlab/bin/gitlab-ctl <stop/start/restart> mattermost` - Stop/start/restart Mattermost.
+- `sudo /opt/gitlab/bin/gitlab-ctl status mattermost` - Check if Mattermost is running. The message printed will start with `run` if it's running or `down` if it's not.
+- `sudo /opt/gitlab/bin/gitlab-ctl tail mattermost` - Watch all Mattermost log files. Press CTRL+C to exit.
```
```suggestion
- `sudo /opt/gitlab/bin/gitlab-ctl tail mattermost` - Watch all Mattermost log files. Press **CTRL+C** to exit.
```

justinegeffen commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
+- `sudo /opt/gitlab/bin/gitlab-ctl status mattermost` - Check if Mattermost is running. The message printed will start with `run` if it's running or `down` if it's not.
+- `sudo /opt/gitlab/bin/gitlab-ctl tail mattermost` - Watch all Mattermost log files. Press CTRL+C to exit.
+- `sudo /opt/gitlab/bin/gitlab-ctl <stop/start/restart/status/tail> <nginx/postgresql/redis/etc>` - Control, view status, or view logs of a different service.
+- `sudo less /var/opt/gitlab/mattermost/config.json` - View the Mattermost config.json directly (arrow keys to scroll, press Q to exit).
```
```suggestion
- `sudo less /var/opt/gitlab/mattermost/config.json` - View the Mattermost `config.json` directly (use the arrow keys to scroll, press **Q** to exit).
```

justinegeffen commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
+- `sudo /opt/gitlab/bin/gitlab-ctl <stop/start/restart/status/tail> <nginx/postgresql/redis/etc>` - Control, view status, or view logs of a different service.
+- `sudo less /var/opt/gitlab/mattermost/config.json` - View the Mattermost config.json directly (arrow keys to scroll, press Q to exit).
+- `sudo gitlab-psql -d mattermost_production` - Access the embedded Mattermost database.
+- `sudo gitlab-rails console production` - Access the GitLab admin console (press CTRL+D to exit).
```
```suggestion
- `sudo gitlab-rails console production` - Access the GitLab admin console (press **CTRL+D** to exit).
```

jasonblais commented:
Thank you @justinegeffen! Addressed feedback. Note that keyboard shortcuts are stylized with capital letters and no bolding: https://docs.mattermost.com/process/documentation-guidelines.html#text-formatting

hmhealey commented on source/process/gitlab-process.rst:
```diff
@@ -29,14 +29,19 @@ If the release contains a security update, PM owner emails @marin and @briann in
+1. Pull the latest master build of GitLab Omnibus, using the `nightly Omnibus packages <https://packages.gitlab.com/gitlab/nightly-builds>`_.
+2. Update the Mattermost version in `config/softwater/mattermost.rb`.
+3. Update the list of Mattermost versions associated with each GitLab release `in the GitLab documentation <https://docs.gitlab.com/omnibus/gitlab-mattermost/#upgrading-gitlab-mattermost>`_, and add a changelog entry.
+4. Build GitLab Omnibus.
```
If you want to include a link to the actual build instructions, they're [here](https://gitlab.com/gitlab-org/omnibus-gitlab/blob/master/doc/build/prepare-build-environment.md), although they require some extra steps.

I have a briefer version with the extra steps [here](https://gist.github.com/hmhealey/b6d3e42a88563ca43f03152e8b86592b#gitlab-omnibus), although they're not exactly detailed

hmhealey commented on source/process/gitlab-process.rst:
```diff
@@ -29,14 +29,19 @@ If the release contains a security update, PM owner emails @marin and @briann in
- - Pre-provisioning OAuth configuration automatically on the Omnibus package
- - Mattermost ChatOps slash command integration
- - OAuth team creation option in GitLab Omnibus
+The Mattermost team then test the upgrade process, and validate the packaging code and OAuth setup which have historically been the main source of issues. Other test areas include:
```
```suggestion
The Mattermost team then tests the upgrade process and validates the packaging code and OAuth setup which have historically been the main source of issues. Other test areas include:
```

hmhealey commented on source/process/support.md:
```diff
@@ -180,13 +180,23 @@ echo | openssl s_client -showcerts -connect <URL>:443 -CApath /etc/ssl/ && echo
-- `gitlab-ctl <stop/start/restart/status/tail> <nginx/postgresql/redis/etc>` - Control, view status, or view logs of a different service.
-- `sudo gitlab-psql -d mattermost_production` - Access the embedded Mattermost database
+Useful commands:
+- `sudo /opt/gitlab/bin/gitlab-ctl reconfigure` - Update `config.json` and other configuration files, then restart all services (including Mattermost).
```
The full path to each of these shouldn't be necessary. `gitlab-ctl` will be on the PATH, so these can just be run as
```suggestion
- `sudo gitlab-ctl reconfigure` - Update `config.json` and other configuration files, then restart all services (including Mattermost).
```
